src/Database/Redis/PubSub.hs

```diff
@@ -83,6 +101,14 @@ sendCmd cmd       = do
     lift $ Core.send (redisCmd cmd : changes cmd)
     modifyPending (updatePending cmd)
 
+totalPendingChanges :: PubSub -> Int
+totalPendingChanges (PubSub{..}) =
+  updatePending subs $ updatePending unsubs $ updatePending psubs $ updatePending punsubs 0
```

Comment:
Yeah, but I am just reusing the existing updatePending code since I initially was trying on leaving any of the existing pub/sub code unchanged.  Probably better as you say to modify it and then update the old pub/sub code to a new internal function.